.github/workflows/rust.yml

```diff
@@ -0,0 +1,36 @@
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    types:
+
+jobs:
+  cargo_nextest:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-22.04-8core
+            flags: ""
+          - os: ubuntu-22.04-8core
+            flags: "--features nightly,test_features"
+          - os: macOS-m1
+            # FIXME: some of these tests don't work very well on MacOS at the moment. Should fix
+            # them at earliest convenience :)
+            flags: "--exclude integration-tests --exclude node-runtime --exclude runtime-params-estimator --exclude near-network --exclude estimator-warehouse"
+    timeout-minutes: 90
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: baptiste0928/cargo-install@21a18ba3bf4a184d1804e8b759930d3471b1c941
```

Comment:
for my education
what is baptiste? 
what is that hash? is that a commit of some version of some binary?

Reply:
`baptiste0928` is a github user, whose github action we’re using: https://github.com/baptiste0928/cargo-install. The hash is indeed the specific commit from the repository. I find it a good practice to lock the actions used to specific commits to avoid an equivalent of the supply chain attacks.